src/types/durableClientTypes.ts

```diff
@@ -1,9 +1,229 @@
-import { FunctionInput } from "@azure/functions";
+import { FunctionInput, HttpRequest, HttpResponse } from "@azure/functions";
+import { EntityId, EntityStateResponse, HttpManagementPayload, IHttpResponse } from "../classes";
+import { DurableOrchestrationStatus } from "../durableorchestrationstatus";
+import { IHttpRequest } from "../ihttprequest";
+import { OrchestrationRuntimeStatus } from "../orchestrationruntimestatus";
+import { PurgeHistoryResult } from "../purgehistoryresult";
 
 export interface DurableClientInput extends FunctionInput {
     type: "durableClient";
 }
 
+/**
+ * Client for starting, querying, terminating and raising events to
+ * orchestration instances.
+ */
+export declare class DurableClient {
+    /**
+     * The name of the task hub configured on this orchestration client
+     * instance.
+     */
+    readonly taskHubName: string;
+
+    /** @hidden */
+    public readonly uniqueWebhookOrigins: string[];
+
```

Comment:
I think this PR should keep the constructor in the types to match the auto-generated types. If we decide to remove the constructor, I think that should be a separate breaking change.

Reply:
This is the type signature of the constructor:

```TS
constructor(
        state: HistoryEvent[],
        instanceId: string,
        currentUtcDateTime: Date,
        isReplaying: boolean,
        parentInstanceId: string | undefined,
        longRunningTimerIntervalDuration: string | undefined,
        maximumShortTimerDuration: string | undefined,
        defaultHttpAsyncRequestSleepTimeMillseconds: number | undefined,
        schemaVersion: ReplaySchema,
        input: unknown,
        private taskOrchestratorExecutor: TaskOrchestrationExecutor
    );
```

Exposing the `constructor()` would also force us to expose other types like `HistoryEvent` (which in turn exposes `HistoryEventType`), `ReplaySchema`, `TaskOrchestrationExecutor` (which in turn also exposes `TaskBase` and other internal classes), etc. In the interest of #352, and in the spirit of this change to let us more intentionally the public APIs, I think we should omit the constructor. Users shouldn't be using this constructor anyway.

Reply:
Sorry, this is actually the signature for the `DurableOrchestrationContext`. Let me add the constructor to the `DurableOrchestrationClient` (the subject of this PR).

Reply:
I see what you're saying, but I still think this is a breaking change and needs its own discussion. Can we add the constructor back for this PR without actually needing to expose all the other types yet? I feel like that should be possible

Reply:
Fair enough. Yeah, it will be possible because this is all in `.ts` files for now anyway.

Reply:
Added the constructor type without exposing all the necessary types too